docs/class-definitions.md

```diff
@@ -1,247 +1,476 @@
+# Space
 
-Default 100ms.
+An instance of a Space created using [spaces.get](#get). Inherits from [EventEmitter](/docs/usage.md#event-emitters).
 
-##### paginationLimit
+## Properties
 
-The number of [history API](https://ably.com/docs/realtime/history) pages searched for the last published cursor position.
+### cursors
 
-Default is 5.
+An instance of [Cursors](#cursors).
 
-### Space
+```ts
+type cursors = instanceof Cursors;
+```
 
-An instance of a Space created using [spaces.get](#getname-options).
+### locations
 
-#### enter(profileData?)
+An instance of [Locations](#locations).
 
-Enter this space with optional profile data and notify other clients in the space via the [membersUpdate](#membersupdate) event.
-This data can be an arbitrary JSON-serializable object which will be attached to the member object and delivered to other members of the space.
+```ts
+type locations = instanceof Locations;
+```
 
-| Property    | Type   |
-| ----------- | ------ |
-| profileData | Object |
+## Methods
 
-#### leave()
+### enter
 
-Leave the space. This removes the member from the space and notifies other space members.
+Enter the space. Can optionally take `profileData`. This data can be an arbitrary JSON-serializable object which will be attached to the [member object](#spacemember). Returns all current space members.
 
-#### on(event, callback)
+```ts
+type enter = (profileData?: Record<string, unknown>) => Promise<SpaceMember[]>;
+```
 
-Used for subscribing to realtime events within the space.
+### leave
 
-##### membersUpdate
+Leave the space. Can optionally take `profileData`. This triggers the `leave` event, but does not immedietly remove the member from the space. See [offlineTimeout](#spaceoptions).
 
-Fires when a member enters or leaves the space. The argument supplied to the event listener callback is the current array of all [SpaceMember](#spacemember) objects within the space.
+```ts
+type leave = (profileData?: Record<string, unknown>) => Promise<void>;
+```
 
-##### enter
+### on
 
-Fires when a member enters the space. The argument supplied to the event listener callback is the [SpaceMember](#spacemember) which entered the space.
+Listen to events for the space. See [EventEmitter](/docs/usage.md#event-emitters) for overloading usage.
 
-##### leave
+Available events:
 
-Fires when a member leaves the space. The argument supplied to the event listener callback is the [SpaceMember](#spacemember) which left the space.
-Note that the leave event will only fire once the [offlineTimeout](#offlinetimeout) has passed.
+- #### _membersUpdate_
 
-#### getMembers()
+```ts
+space.on('membersUpdate', (members: SpaceMember[]) => {});
+```
 
-Returns an array of all [SpaceMember](#spacemember) objects currently in the space, including any members who have left and not yet timed out. (_see: [SpaceOptions.offlineTimeout](#offlinetimeout)_)
+Listen to updates to members:
```

Comment:
`SpaceMember` is the name of the type used internally, and it's different from presence. I think members is just used in general, but it should be equivalent to `SpaceMember`. What would you suggest? Not sure if we really need to use the whole type name everywhere, we don't do this here or in ably docs.

Reply:
Yeah, I think it's more we should be consistent how we refer to it, otherwise it can seem like we're talking about two different types. Even if we explain upfront that SpaceMember is the type and that means you're a member of a space.

Reply:
I'm going to not do this change as I think this would require a lot of rewriting, and I'm still not sure how it should executed. For example we use "members" but we can't just use SpaceMembers**s** as that would be a different type. If it's ok, I'll leave this to you @m-hulbert when these changes and API design changes are in.